src/main/antlr/org/hibernate/sqm/parser/hql/internal/antlr/HqlParser.g4

```diff
@@ -87,10 +88,26 @@ orderingSpecification
 	;
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// LIMIT/OFFSET clause
+
+limitClause
+	: LIMIT parameterOrNumberLiteral
+	;
+
+offsetClause
+	: OFFSET parameterOrNumberLiteral
+	;
+
+parameterOrNumberLiteral
+	: parameter
+	| INTEGER_LITERAL
+	;
```

Comment:
I'm not sure we want to support parameters here.  Not all databases support parameterized limit/offset, even if they support limit/offset at all.